### DISCUSSION TITLE
Add the "Special lift equipment" - creation of hidden weapons, lifting devices, hoists of ammunition, etc.

### DISCUSSION THREAD

**Discussion by MaaiG** [FakeFishGames/Barotrauma#7432] [Ideas and suggestions]
Special lift equipment must be added to the game!

**Сurrent problems!**
Today in the game you can create "elevator systems" for vertical and horizontal movement as added submarines:
- `Docking ports` - as endpoints.
- `Pumps` and `Engines` - like motors vertical and horizontal movement.
For exemple:

https://user-images.githubusercontent.com/93348499/142764030-95aa7e9e-4d56-45ce-ba31-020be437ab21.mp4

https://user-images.githubusercontent.com/93348499/142764043-46fdbabd-1b77-44d4-85da-0680222e0e53.mp4

https://user-images.githubusercontent.com/93348499/142764060-32baadef-601b-4089-9eca-f4822c61e7d6.mp4

But this method is like a "Duct tape" and has a lot of complications and limitations:
1. `Hull` of the added submarine cannot be located or moved through the `Hull` of the main submarine. For this reason `Elevator engine` (`Hull` with pump and docking ports) should be located at a distance from `Elevator` itself  (there is no `Hull` and it can move inside your submarine’s `Hulls`)
2. Accordingly, it is possible to move only in a separate elevator shaft which needs to be placed on the side of the submarine if your "elevator" passes through the entire submarine.
4. `Docking ports` can be connected on two sides and the first connection is remembered. This leads to several problems which are difficult to solve with when there are more than three levels (the symmetry problem and others). 
5. ~~`Elevator engine` cannot be far from `Elevator`. Otherwise it becomes invisible!~~
Why? [Look here](https://github.com/Regalis11/Barotrauma/issues/7402).
6. Due to point 1, you cannot create `Outer hatches` of moving guns because the hatch not overlaying `hull` is not an obstacle to monsters.
7. No hard binding. The speed of the elevator depends on the speed of the submarine. If the submarine quickly dives, your elevator cannot descend etc.

It would be good if the game added objects similar to symbiosis `docking ports` and `way points`. Such objects will be able to set the end points and intermediate points of movement between them. A other object (attached to your added submarine) will move between them at a specified speed when receiving a signal from your connected buttons.

This will solve many problems, will remove current restrictions and will allow to build many interesting submarines, forposts, etc. :+1:

**Comment by mLuby** [comment]
Just to add an alternative, you can make lifts/elevators out of vanilla hatches—no special hulls or shuttles necessary.

![ezgif com-optimize-2](https://user-images.githubusercontent.com/2483420/143277914-ef9edb5e-a7d8-4715-ba54-333c7465bb8f.gif)

**Reply by Ek-Videogames** [reply]
> Just to add an alternative, you can make lifts/elevators out of vanilla hatches—no special hulls or shuttles necessary.

Sadly this method does not play nice with netcode anymore and will just send items into the shadow realm or worse crush your character into the 4th dimension. ( I feel like because the devs changed how doors interact with ragdolls )

- also the hull implementation would be about 2000% more useful and repurposeable for all kinds of things.

**Comment by MaaiG** [comment]
Always, I try to minimize the number of used items! And improve logical systems to reduce the number of logical components.
You have a cool system! )) Bit slow and loads the knee joints ;) but very cool! )

I want the developers to add a mechanism to the game that is specifically designed for this purpose!

**Comment by WickedWolfy** [comment]
+1

**Comment by KB0MB4** [comment]
Have to mention this https://steamcommunity.com/sharedfiles/filedetails/?id=2440212300
Kinda broken after hull placement was updated, gets water everywhere, still works as a concept.
Want to add another point for ingame ready elevators, bots cant properly use editor made ones.
Also we cant dock two subs with more than one docking port so bots could understant the waypoint links, that means double sided elevator is not possible.

So please implement this, i'm really not a fan of 50 meter tall ladders.
